src/mpid/ch4/netmod/ofi/ofi_am_impl.h

```diff
@@ -351,38 +339,39 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_do_am_isend(int rank,
         need_packing = true;
     }
 
+    if (MPIDI_OFI_global.deferred_am_isend_q) {
+        /* if the deferred queue is not empty, all new ops must be deferred to maintain ordering */
+        goto fn_deferred;
+    }
```

Comment:
If we add a flag to indicate that we can skip this deferred check, then `issue_deferred_am_isend_eager` can directly call this function for the eager part, avoid all the code duplication, right?

Reply:
I don't think that is a good idea. 1. It is not clear why the function has a parameter for skipping this checking by itself. The one look at the function must knows that the progress engine would call it for issuing deferred operation. 2. It is not just that one check need to be skipped, it also needs to skip the part of enqueuing when a pack buffer cannot be allocated which is a further complication of the logic. 3. If the progress engine would this function, it would need a way to know that it actually succeeded in issuing the operation so that the progress engine can dequeue and free the deferred operation. But since the function return `mpi_errno`, I am not sure if we want to add a new return value for "message not issued".

Reply:
They are merged.